File: C  programs/Problems on Recursion/Pattern Printing/Pattern4.c
/* Write a recursive program which display below pattern. 

Output : A B C D E F

*/

#include<stdio.h>

void DisplayI()
{
	int iCnt=0;
	char ch='A';
	for(iCnt='A';iCnt<='E';iCnt++)
	{
		printf("%c\n",ch);
		ch++;
	}	
}

void DisplayR()
{
	static int iCnt='A';
	char ch='A';
	if(iCnt<='E')
	{
		printf("%c\t",iCnt);
		iCnt++;
		DisplayR();
	}	
}

int main()
{
   //DisplayI();
   DisplayR();
   return 0;
}

File: Java programs/Problems on Pattern Printing/String Pattern/Pattern4.java
/* Write a java program which accept String from user and display below pattern.

Input : Hello
Output :  H  e  l  l  o
          H  e  l  l  *
		  H  e  l  *  *
		  H  e  *  *  *       
		  H  *  *  *  *
          H  e  *  *  *		   
          H  e  l  *  *
		  H  e  l  l  *
		  H  e  l  l  o                          

*/ 
		             
import java.util.*;

class Pattern
{
    public void Pattern(String str)
	{
        char Arr[]=str.toCharArray();
	    int i=0,j=0;
		int iSize=Arr.length;
	    
	    for(i=Arr.length-1;i>=0;i--)        
	    {
			for(j=0;j<Arr.length;j++)
		    {
			    if(i>=j)
			    {
			      System.out.print(Arr[j]+"\t");
			    }
				else
				{
					System.out.print("*"+"\t");
				}	
		    }	
		    System.out.println();
        }
		
		for(i=1;i<Arr.length;i++)        
	    {
			for(j=0;j<Arr.length;j++)
		    {
			    if(i>=j)
			    {
			      System.out.print(Arr[j]+"\t");
			    }
				else
				{
					System.out.print("*"+"\t");
				}	
		    }	
		    System.out.println();
        }
	}
}

class Lba394
{
	public static void main(String arg[])
	{
		Scanner sobj=new Scanner(System.in);
		
		System.out.println("Enter string");
		String str=sobj.nextLine();
		
		Pattern pobj=new Pattern();
		pobj.Pattern(str);
	}	
}
